financialaid/views.py

```diff
@@ -224,23 +227,56 @@ class FinancialAidActionView(UpdateAPIView):
     queryset = FinancialAid.objects.all()
 
 
-class GetLearnerPriceForCourseView(APIView):
+class CoursePriceListView(APIView):
     """
-    View for retrieving a leaner's price for a course run
+    View for retrieving a learner's price for course runs in all enrolled programs
+    """
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        """
+        GET handler
+        """
+        user = request.user
+        program_enrollments = ProgramEnrollment.objects\
+            .select_related('user', 'program')\
+            .filter(user=user, program__live=True).all()
+        formatted_course_prices = []
+        for program_enrollment in program_enrollments:
+            try:
+                response_dict = get_formatted_course_price(program_enrollment)
+                response_dict['program_id'] = program_enrollment.program.id
+                formatted_course_prices.append(response_dict)
+            except ImproperlyConfigured:
```

Comment:
I'm not sure we want to silence this exception here. If `get_formatted_course_price` doesn't have a valid course price to choose from then something's wrong with our configuration. I think it would be better to have the exception cause a 500 error here